### PR TITLE
feat: support split packages

### PR DESCRIPTION
**Pull request by coryan**
With this change package maintainers can split `google-cloud-cpp` into smaller packages. Some package ecosystems have constraints on the build time for the packages, or on the size of the packages. `google-cloud-cpp` is exceeding some of these limits (notably build times on Conda). The solution is to create a smaller package with the core libraries, and then one or more packages with a subset of the features built on top of this core.

In addition to the necessary support in CMake, this introduces a build to test this configuration.

Motivated by conda-forge/google-cloud-cpp-feedstock#141

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12049)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -155,6 +155,17 @@ GOOGLE_CLOUD_CPP_OVERRIDE_GOOGLEAPIS_URL for details.
 ]==])
 mark_as_advanced(GOOGLE_CLOUD_CPP_OVERRIDE_GOOGLEAPIS_URL_HASH)
 
+option(
+    GOOGLE_CLOUD_CPP_USE_INSTALLED_COMMON
+    [==[
+Use a pre-installed version of the common libraries and proto files.
+
+This option is required to support package maintainers that split
+`google-cloud-cpp` into multiple (hopefully smaller) packages.
```
Comment (devbww) [inline CMakeLists.txt:164]: Is it even possible to split into larger packages?

**Review by devbww — APPROVED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -231,12 +241,32 @@ cmake_dependent_option(
...
+add_subdirectory(external/googleapis)
 add_subdirectory(google/cloud)
+
+if (GOOGLE_CLOUD_CPP_USE_INSTALLED_COMMON)
+    find_package(google_cloud_cpp_common CONFIG REQUIRED)
+    if (GOOGLE_CLOUD_CPP_ENABLE_GRPC)
+        find_package(google_cloud_cpp_grpc_utils CONFIG REQUIRED)
+        find_package(google_cloud_cpp_googleapis CONFIG REQUIRED)
+        find_package(google_cloud_cpp_iam CONFIG REQUIRED)
+        find_package(google_cloud_cpp_googleapis CONFIG REQUIRED)
```
Comment (dbolduc) [inline CMakeLists.txt:260]: Repeated command

**Review by dbolduc — COMMENTED** [review]
(not a full review)

File: CMakeLists.txt (hunk 1 above)
Comment (coryan) [reply CMakeLists.txt]: Fixed.

File: CMakeLists.txt (hunk 2 above)
Comment (coryan) [reply CMakeLists.txt]: Removed and removed an unneeded command too

**Pull request merged**